cli/Cargo.toml

```diff
@@ -11,6 +11,14 @@ selfmanage = []
 end-to-end-tests = []
 extensions = ["dep:deno_runtime", "dep:deno_core", "dep:deno_ast"]
 
+[[test]]
+name = "extensions"
+path = "tests/extensions/mod.rs"
+
+[[test]]
+name = "end_to_end"
+path = "tests/end_to_end/mod.rs"
+
```

Comment:
@cd-work and I discussed this and many other options when the `end_to_end` integration test was first added. Using this instead of auto-discovery is confusing. Also, this makes it more difficult for the tests to share code since you can no longer just put `mod common;` in the top-level file and then `use crate::common;` where necessary.

Reply:
I mean at the very least with this approach there wouldn't be a reason to separate end-to-end and other integration tests? That should make sharing a little easier at least.

Reply:
What would be the ideal structure in place of this? A couple options I could see:

- Single top-level integration tests file which includes test sub-modules (has the drawback of having everything together, if this is undesirable)
- Separate but discoverable `extensions.rs` + `extensions/<module>.rs` and `end_to_end.rs` + `end_to_end/<module.rs>` (has the drawback of warnings in common code)